test/test_roscpp/test/CMakeLists.txt

```diff
@@ -1,6 +1,6 @@
 catkin_add_gtest(${PROJECT_NAME}-test_version test_version.cpp)
 if(TARGET ${PROJECT_NAME}-test_version)
-  target_link_libraries(${PROJECT_NAME}-test_version)
+  target_link_libraries(${PROJECT_NAME}-test_version ${catkin_LIBRARIES})
```

Comment:
I could not build and run the unit tests without the following changes.

Reply:
Did you end up just running the tests on Linux but with a build where `HAVE_EPOLL` was forced to 0?

Reply:
That's correct yes.

Reply:
And prior to this fix the tests failed to complete successfully?

Reply:
Yes, sorry, I should have mentioned that. 

Also, I am looking at the current build failure.

Reply:
My impression is that that test has been flakey for a while— it definitely has a timing sensitivity and does _sometimes_ pass on the buildfarm. However, if you have further insights on it, that would be great.